nifi-docker/docker/Dockerfile

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements. See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership. The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM java:8u91-jdk
+MAINTAINER Apache NiFi <dev@nifi.apache.org>
+
+ARG UID
+ARG GID
+ARG NIFI_VERSION
+ARG NIFI_BINARY
+
+ENV NIFI_BASE_DIR /opt/nifi
+ENV NIFI_HOME $NIFI_BASE_DIR/nifi-$NIFI_VERSION
+
+# Setup NiFi user
+RUN groupadd -g $GID nifi || groupmod -n nifi `getent group $GID | cut -d: -f1`
+RUN useradd --shell /bin/bash -u $UID -g $GID -m nifi
+RUN mkdir -p $NIFI_HOME 
+
+ADD $NIFI_BINARY $NIFI_BASE_DIR
```

Comment:
I did a little research and your right. It will have to download the binary since the built binary would not be referenceable. I don't think it would be the end of the world to have 2 Dockerfiles. Really the files could be identically aside from the piece that pulls in the binary. This one could be used for people doing local builds and development while the one that downloads the binary could be used for pushes to Dockerhub. I'll work something up for that here shortly and make another commit.

Thanks everyone for chiming in on this. There are a lot of ways to do this I know and things keep changing a little bit but we will get there =)